### DISCUSSION TITLE
Possible new serial number for Libre2

### DISCUSSION THREAD

**Discussion by aslepykh** [NightscoutFoundation/xDrip#3439] [Q&A]
I suppose new serial number for Libre2 (RU region) is released. Can't pair it with xdrip+ (3e86af7-2023.11.14) on Pixel 5.
OOP2 (8a0ee8f-2022.06.10) is installed also.

Libre2 (RU) was activated on another device (iPhone 13 Libre2 App). This device was disabled during pairing attepts with Pixel 5.

https://github.com/NightscoutFoundation/xDrip/blob/10f2319be05be5a1f73ad87ff3a2b2c900272818/app/src/main/java/com/eveningoutpost/dexdrip/models/LibreOOPAlgorithm.java#L304-L305

![Screenshot_20240428-143342](https://github.com/NightscoutFoundation/xDrip/assets/145323274/64e1fb3a-6ee1-4561-bfd3-5dcb10070085)

**Comment by Navid200** [comment]
I converted the issue into a discussion since you seem to have missed our request not to open an issue.
Let's find out exactly what's going on before turning this into an issue.

**Reply by aslepykh** [reply]
My bad!
What additional information should I give you about this problem?

**Reply by Navid200** [reply]
@PhilTU

**Comment by PhilTU** [comment]
> OOP2 (8a0ee8f-2022.06.10) is installed

That's old. You want May 2023 version.

May not resolve it but let's isolate that first

**Reply by aslepykh** [reply]
So I found and installed OOP2 (27da3f5-2023.05.25).

Issue isn't resolved

**Reply by PhilTU** [reply]
Is error log identical - flipping to Libre 1 as not recognised?

See Tzachi response below.

Libre 2+ and RU version are like US Libre 2 that we haven't been supporting.

**Reply by aslepykh** [reply]
Yeah, error log is identical. Trying out Tzachi recommendations right now

**Comment by tzachi-dar** [comment]
Here is a new version of xdrip: https://drive.google.com/file/d/1i2n6zNFrNGcaRSVOO58dX7jzuEWfRx2g/view?usp=sharing
And oop2: https://drive.google.com/file/d/1PbQQoY_KfOrjOboLgKrp4aQy3HZzvZRf/view?usp=sharing

You need to create a backup of xdrip settings and DB and then restore them. You will not be able to restore to google drive, so please backup them up manually.

Please install them and let me know how it goes. I'll soon be offline for 24 hours but hopefully it will work.

**Reply by aslepykh** [reply]
So I reinstalled both apps. Here's log

I tried multiple times scanning, same result

![Screenshot_20240428-185735](https://github.com/NightscoutFoundation/xDrip/assets/145323274/26240d88-f83c-401d-8d9c-cfc12785a3d3)
![Screenshot_20240428-185745](https://github.com/NightscoutFoundation/xDrip/assets/145323274/2339ccf1-1cb0-4d66-afa4-7a5490cf4e8e)

**Comment by tzachi-dar** [comment]
I'll take a look tomorrow, but it might be a new sensor and not just a new
sn.

Does the box say libre 2?

בתאריך יום א׳, 28 באפר׳ 2024, 19:02, מאת aslepykh ***@***.***

**Reply by aslepykh** [reply]
Yes, it's Libre 2 for sure. 
I'll be waiting, thanks!

**Reply by PhilTU** [reply]
From elsewhere I was aware there is an issue with a modified L2 sensor for Russian market as Diabox dev is "working on it".

A haha guidos has written at the same time as me.

**Reply by aslepykh** [reply]
Does that mean that there is no simple solution?

**Reply by parapenT1sta** [reply]
Libre 2 C6 was added to xDrip4iOS: https://github.com/JohanDegraeve/xdripswift/pull/539/commits/549647a7cd5137e6a6fc9319887473736a1912ca

**Reply by tzachi-dar** [reply]
I'll try to take a look tonight, but since I'm traveling tomorrow, this might have to wait.

**Comment by PhilTU** [comment]
> Does that mean that there is no simple solution?

Exactly. We don't have support for US or Canada Libre 2 sensors (or Aus / NZ) so Russia is joining that list (UK too is about to get Libre 2+ with same issue).

**Reply by aslepykh** [reply]
Got it. 

I have managed to get it to work with Juggluco, but it has some shortcomings compared to xdrip+, but it works for me.

Thanks for the help and clarification!

**Comment by tzachi-dar** [comment]
I have created oop2 version for the c6 libre2.
Please note that we are using the first 3 numbers so I might need more information for this. Right now it is 0xc60930

oop2: https://drive.google.com/file/d/1z4IzWla5nor2JJbXVglWhhWXUG_nMmok/view?usp=sharing

The xdrip version is:

**Comment by tzachi-dar** [comment]
I have created oop2 version for the c6 libre2.
Please note that we are using the first 3 numbers so I might need more information for this. Right now it is 0xc60930

oop2: https://drive.google.com/file/d/1z4IzWla5nor2JJbXVglWhhWXUG_nMmok/view?usp=sharing

The xdrip version is:
https://drive.google.com/file/d/14NOgsjTTt0jN2TJ5yWICUZtiJc9254w4/view?usp=sharing

**Reply by Navid200** [reply]
Do we need a PR for this?

**Reply by PhilTU** [reply]
I have a Libre 2+ sensor (UK) arriving today and will use the above test version with it. 
![fsl2plus-sensor-card](https://github.com/NightscoutFoundation/xDrip/assets/76240903/39237d98-cfe4-485e-b680-63c31ad6dca9)

these are going to be the standard Libre 2 offerring in the UK from next month via prescriptions and are already the retail version online from Abbott. It has a 15 day life - not sure if that needs to be accommodated explicitly in xDrip+

**Reply by PhilTU** [reply]
> I have created oop2 version for the c6 libre2. Please note that we are using the first 3 numbers so I might need more information for this. Right now it is 0xc60930
> 
> oop2: https://drive.google.com/file/d/1z4IzWla5nor2JJbXVglWhhWXUG_nMmok/view?usp=sharing
> 
> The xdrip version is: https://drive.google.com/file/d/14NOgsjTTt0jN2TJ5yWICUZtiJc9254w4/view?usp=sharing

I was unsuccesful with a UK Libre 2plus sensor and these files.patchinfo reports as C6 09 31 @tzachi-dar

**Reply by PhilTU** [reply]
I have a 2+ sensor on now and there are 10 days left when I can test any changes. @tzachi-dar @Navid200 should I make this an "issue" as currently xDrip cannot work with the soon to be default UK Libre 2plus sensor. patchinfo reports as C6 09 31 and the above test versions were coded for  C6 09 30

**Reply by Navid200** [reply]
@PhilTU Is this not already addressed by this open issue?  https://github.com/NightscoutFoundation/xDrip/issues/2725

**Reply by PhilTU** [reply]
No. That issue is specifically the misrepresentation of the serial number by xDrip when a sensor with serial 3HMxxxx is shown on the screen as 0HMxxxx

This is in some way a cosmetic issue except when debugging it is useful to know if the sensor is a Libre 1 (starts zero) or a Libre 2 (starts with a 3).

My current problem relates to the new sensor Libre 2 Plus not being recognised correctly and hence defaulting to be handled as a Libre 1 which doesn't work.

**Reply by Navid200** [reply]
@PhilTU Go ahead and open an issue.

**Comment by tzachi-dar** [comment]
@PhilTU 
Here are the files for c6-09-31 series:
xdrip: https://drive.google.com/file/d/1dkhhgfNLWxvQ8O7GbePffjG9BtajLhcy/view?usp=sharing
oop2: https://drive.google.com/file/d/15W6L3w3dwCNzESoBO2mjfsWmZafw9s73/view?usp=sharing
Let me know how it works.

**Reply by PhilTU** [reply]
Thanks. This connected to the sensor ok and is receiving values via Bluetooth for a few minutes now.

There was a a couple of error messages in the log 

DexCollectionService
06:40 Go null pointer trying to set CCCD descriptor
06:40 gegorybel: force disconnect!

have received 9 values with 4 missed so far

**Reply by PhilTU** [reply]
working well for >2 hours

**Reply by PhilTU** [reply]
with the exception of one loss of signal at around 12h use it's been solid for 24h.

**Comment by tzachi-dar** [comment]
I have created a PR for the fix: https://github.com/NightscoutFoundation/xDrip/pull/3514

**Comment by Andrey997** [comment]
If there are other ideas for Libre 2 (region RU), I may test it out by wearing it all the time.

**Comment by yafoxins** [comment]
@tzachi-dar Is there any possibility to come up with something for Libre 2 RU?
